### PR TITLE
Create a Data model

### PR DESCRIPTION
**Pull request by Wahaj-Ali**
## Highlights
- Create a new Rails project.
- Make sure the PostgreSQL database is set up.
- Use database migration files to maintain database schema.

### CODE REVIEW

File: db/migrate/20230328220723_create_post.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+
+      t.timestamps
+    end
+  end
+end
```
Comment (alexander16108) [inline db/migrate/20230328220723_create_post.rb:8]: as required you should have a migration file for creating a posts table as seen below 
image

File: db/migrate/20230328223126_create_user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateUser < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+
+      t.timestamps
+    end
+  end
+end
```
Comment (alexander16108) [inline db/migrate/20230328223126_create_user.rb:8]: as required you should have a migration file for creating a users table as seen below
image

File: db/migrate/20230328222923_create_like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateLike < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+
+      t.timestamps
+    end
+  end
+end
```
Comment (alexander16108) [inline db/migrate/20230328222923_create_like.rb:8]: as required you should have a migration file for creating a like table as seen below
image

File: db/migrate/20230328223324_create_comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class CreateComment < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+
+      t.timestamps
+    end
+  end
+end
```
Comment (alexander16108) [inline db/migrate/20230328223324_create_comment.rb:8]: as required you should have a migration file for creating a comment table as seen below
image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Wahaj-Ali , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230328220723_create_post.rb (hunk 1 above)
Comment (Wahaj-Ali) [reply db/migrate/20230328220723_create_post.rb:8]: @alexander16108 I created the post table and then used other commands to add it to this table. You may find migration files in my 'db' folder. Please have a re-look.

Comment (alexander16108) [reply db/migrate/20230328220723_create_post.rb:8]: hi @Wahaj-Ali i apologise for not cross checking. submit it and it will be approved

Comment (Wahaj-Ali) [reply db/migrate/20230328220723_create_post.rb:8]: @alexander16108 thank you for your cooperation 😄

**Review by Reem-lab — COMMENTED** [review]
Hi @Wahaj-Ali,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't follow :

- [ ] the right Gitflow since your PR should be from a new branch to dev not from dev to main so kindly follow the right Gitflow you can read more about it in this [artice](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) 🤩✅💯💯🚀🚀


- [ ] I think you have a live demo section in your README file and to have a professional README file you should remove this section till you have your app deployed so you can just add the live demo section in case you have deployed link for your project is present. .✅❗💯🤩✨✨

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Wahaj-Ali** [comment]
@Reem-lab thank you so much for pointing out this huge error. I will follow Gitflow.

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @Wahaj-Ali   👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

**Kindly note:**  When you submit a request for s second opinion, you are not supposed to merge/ close the PR or request another review in your dashboard (As I have realized you just did) else this will alter the results of the second opinion.


That said, I have inspected your project and the previous code reviewer (@alexander16108 ) might have made a mistake indeed your project was well implemented.

However, as mentioned by the reviewer @Reem-lab , your project is not professional and the flow used is wrong. You currently used GitHub flow and the flow that is required is Gitflow. Kindly have a look at the resource shared by @Reem-lab for some context.

For this reason, you will need to create a new pull request and create it merging it into dev and not main.

In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Pull request closed without merging**